src/Plugin/search_api/processor/EntityReferenceWithUri.php

```diff
@@ -0,0 +1,195 @@
+        'type' => 'string',
+        'processor_id' => $this->getPluginId(),
+        'settings' => [
+          'filter_terms' => [],
+          'target_type' => $field->getSetting('target_type'),
+        ],
+        'is_list' => TRUE,
+      ];
+      $fieldname = 'entity_reference_with_term__' . str_replace('.', '__', $field->id());
+      $properties[$fieldname] = new EntityReferenceWithUriProperty($definition);
+    }
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addFieldValues(ItemInterface $item) {
+    // Skip if no Entity Reference with URI fields are configured.
+    $relevant_search_api_fields = [];
+    $search_api_fields = $item->getFields(FALSE);
+    foreach ($search_api_fields as $field) {
+      if (substr($field->getPropertyPath(), 0, 28) == 'entity_reference_with_term__') {
+        $relevant_search_api_fields[] = $field;
+      }
+    }
+    if (empty($relevant_search_api_fields)) {
+      return;
+    }
+
+    $content_entity = $item->getOriginalObject()->getValue();
+    $type_and_bundle_prefix = $content_entity->getEntityTypeId() . '__' . $content_entity->bundle() . '__';
+    foreach ($relevant_search_api_fields as $search_api_field) {
+      $entity_type = $search_api_field->getConfiguration()['target_type'];
+      $filter_terms = $search_api_field->getConfiguration()['filter_terms'];
+      $filter_tids = array_map(function ($element) {
+        return $element['target_id'];
+      }, $filter_terms);
+      $require_all = $search_api_field->getConfiguration()['require_all'];
+      $field_name = substr($search_api_field->getPropertyPath(), 28);
+      $field_name = str_replace($type_and_bundle_prefix, '', $field_name);
+      if ($content_entity->hasField($field_name)) {
+        $referenced_terms = [];
+        foreach ($content_entity->get($field_name)->getValue() as $values) {
+          foreach ($values as $value) {
+            // Load the entity stored in our field.
+            $target_entity = \Drupal::entityTypeManager()
+              ->getStorage($entity_type)
+              ->load($value);
+            // Load the taxonomy terms on the entity stored in our field.
+            $referenced_terms = array_merge($referenced_terms, array_filter($target_entity->referencedEntities(), function ($entity) {
+              if ($entity->getEntityTypeId() != 'taxonomy_term') {
+                return FALSE;
+              }
+              else {
+                return TRUE;
+              }
+            }));
+          }
+
+          $referenced_tids = array_map(function ($element) {
+            return $element->id();
+          },
+            $referenced_terms
+          );
+          if ($require_all) {
+            if (count(array_intersect($referenced_tids, $filter_tids)) == count($filter_tids)) {
+              // "All" and all the terms specified are present.
+              $label = $target_entity->label();
```

Comment:
This looks to be outside of the `foreach` where it's instantiated and could be possibly be undefined?